docs/modeling/scripts.md

```diff
@@ -0,0 +1,152 @@
+---
+Title: Scripts
+---
+
+# Scripts 
+Scripts are used to execute a script as part of a process. Process variables can be passed to the script and the results of a script can be sent back to a process instance as process variables.
+
+Scripts are executed outside of the [runtime bundle](../architecture/application.md#runtime-bundle) by the [script runtime service](../architecture/application.md#script-runtime-service) with the results being returned to the process after execution. Script design uses the functionality of  [Monaco](https://github.com/Microsoft/monaco-editor) and the [Graal javascript engine](https://github.com/graalvm/graaljs) for execution. 
+
+Scripts can be added to a process definition by using a [script task](../modeling/processes/bpmn/script.md).
+
+The properties of a script are:
+
+| Property | Description | Example | Required |
+| -------- | ----------- | ------- | -------- |
+| `Name` | The name of the script. Script names must be in lowercase and between 1 and 26 characters in length. Alphanumeric characters and hyphens are allowed, however the name must begin with a letter and end alphanumerically. | order-script | Yes |
+| `Language` | The development language the script is written in | Javascript | Yes | 
+
+## Modeling scripts
+
+### Variables
+There are two types of variables associated with a script. Script variables are stored in the `<script-name>-extensions.json` file and declared variables are declared as part of the script itself.
+
+#### Script variables
+Script variables are used to pass values between a script and a process using script variables and [process variables](../modeling/processes/variables.md) with the mapping between the two configured in the [script task](../modeling/processes/bpmn/script.md) itself. Script variables can be configured using the **Edit Script Variables** button in the properties panel of the GUI or the JSON editor. 
+
+The following are the data types that script variables can be set as:
+
+| Type | Description | Example | 
+| ---- | ----------- | ------- |
+| String | A sequence of characters | `#Mint-Ice-Cream-4!`
+| Integer | A positive whole number | `642` |
+| Boolean | A value of either `true` or `false` | `true` |
+| Date | A specific date in the format `YYYY-MM-DD` | `2020-04-22` | 
+| Datetime | A specific date and time in the format `YYYY-MM-DD HH:mm:ss` | `2020-09-10 22:30:00`
+| File | A [file](../modeling/files.md) uploaded into a process definition or as part of a process instance or task | 
+| JSON | A JSON object | `{"flavor" : "caramel"}` | 
+
+Script variables are stored in the `variables` of the `<script-name>-extensions.json` and can also be viewed through the UI in the **Extensions Editor**.
```

Comment:
In my opinion it should be **Extensions Editor** given the name of the file is the same as for processes, but I will update it to metadata to match the UI until this can be reviewed at the next release.